packages/x-license/src/generateLicense/generateLicense.ts

```diff
@@ -8,6 +9,7 @@ const licenseVersion = '2';
 export interface LicenseDetails {
   orderNumber: string;
   expiryDate: Date;
+  purchaseDate?: Date;
```

Comment:
The generation is done through a form on Toolpad
Don't you think that it would be easier and more reliable to always pass the argument saying which version of the plan we want to apply instead of having both a param and some internal logic?

And I don't think we want to store the date in the license since for the same date we can have people that are grandfathered and people that are not.
IMHO it's simpler to store a version of the plan (either 1-2-3, or some other name representing the version).

Reply:
we could do the 'plan version' approach. I thought of that as well. The problem with that is that we basically hardcode everything that way and its less flexible.

Take a case where the license generation has a bug that stems from a new scope being passed in. We would have to add another scope to fix that. I will explore this nonetheless.

Grandfathered licenses should not be created with a `purchaseDate`.

Let me explore a bit more. This was meant as a first iteration anyways! :P

Reply:
> Grandfathered licenses should not be created with a purchaseDate.

Isn't it super arbitrary?
How does it work when we change the pricing again and we have some grandfathered licenses that need the `purchaseDate` to access the charts & tree view and some licenses for some v3 format that need to have specific rules.

_____

The problem with the timestamp is that it does not represent the condition of application, and since the condition does not really depend on the type, but more on the contract signed by the customer, I don't think it's a good way of representing this data.

Reply:
make sense ... I'll take another route ... thanks for the feedback!